### PR TITLE
finished apt_game tests & mypy/flake8/linter checked

### PR DESCRIPTION
**Pull request by Yuhu-kth**
Seems there is a bug here in file apt_game_util.py
function p_o_given_b_a1_a2 
line 253 `prob += b[s] * config.T[a1][a2][s][s_prime] * config.Z[a1][a2][s_prime][o]`
Z is observation tensor with the shape of |S|x|O|

### CODE REVIEW

**Comment by Limmen** [comment]
@Yuhu-kth Yes you are right, good catch. You can remove that function completely, i.e., the function 'p_o_given_b_a1_a2' it is never used.

File: simulation-system/libs/gym-csle-apt-game/tests/test_apt_game_util.py [hunk 1]
```diff
@@ -13,3 +18,263 @@ def test_b1(self) -> None:
...
+            config=example_apt_game_config,
+            a2=1,
+            s=0,
+        )
+        assert isinstance(example_next_belief, np.ndarray)
+        assert sum(example_next_belief) == 1
+
+    def test_sample_attacker_action(
+        self, example_attacker_strategy: MultiThresholdStoppingPolicy
+    ):
```
Comment (Limmen) [inline simulation-system/libs/gym-csle-apt-game/tests/test_apt_game_util.py:182]: --> None

File: simulation-system/libs/gym-csle-apt-game/tests/test_apt_game_util.py [hunk 2]
```diff
@@ -13,3 +18,263 @@ def test_b1(self) -> None:
...
+            p_a,
+            save_dir,
+            checkpoint_traces_freq,
+            gamma,
+        )
+        o = 1
+        b = [0.5, 0.5]
+        a1 = 0
+        a2 = 1
+        print(example_apt_game_config.S)
```
Comment (Limmen) [inline simulation-system/libs/gym-csle-apt-game/tests/test_apt_game_util.py:234]: Remove prints

**Review by Limmen — CHANGES_REQUESTED** [review]
Looks good, minor things to fix, then I will merge it.

File: simulation-system/libs/gym-csle-apt-game/tests/test_apt_game_util.py (hunk 2 above)
Comment (Yuhu-kth) [reply simulation-system/libs/gym-csle-apt-game/tests/test_apt_game_util.py]: fixed

**Review by Limmen — APPROVED** [review]

**Pull request merged**